File: Maximum Absolute Sum of Any Subarray.py
class Solution(object):
    def maxAbsoluteSum(self, nums):
        currentmax=0
        currentmin=0
        maxsum=0
        minsum=0
        for i in range(len(nums)):
            currentmax=max(nums[i],currentmax+nums[i])
            maxsum=max(currentmax,maxsum)

            currentmin=min(nums[i],currentmin+nums[i])
            minsum=min(currentmin,minsum)

        return(max(maxsum,abs(minsum)))

File: Java/Maximum Absolute Sum of Any Subarray.java
class Solution {
    public int maxAbsoluteSum(int[] nums) {
        int curr=nums[0];
        int maxcurr=nums[0];
        int ncurr=nums[0];
        int mincurr=nums[0];
        int total=nums[0];
        for(int i = 1 ; i < nums.length ; i++){
            curr=Math.max(curr+nums[i], nums[i]);
            maxcurr=Math.max(curr,maxcurr);
            ncurr=Math.min(ncurr+nums[i], nums[i]);
            mincurr=Math.min(ncurr,mincurr);
            total+=nums[i];  
        }
        return Math.max(maxcurr,Math.abs(mincurr));
        
    }
}
